pkg/thanos/statefulset.go

```diff
@@ -548,3 +559,62 @@ func prefixedName(name string) string {
 func volumeName(name string) string {
 	return fmt.Sprintf("%s-data", prefixedName(name))
 }
+
+func generateRemoteWriteConfigYaml(remoteWrites []monitoringv1.RemoteWriteSpec) yaml.MapSlice {
```

Comment:
I expect that we could reuse the existing code from [pkg/prometheus/promcfg.go](https://github.com/prometheus-operator/prometheus-operator/blob/9c7250b81454279301dacc0fa413ef1be33c12df/pkg/prometheus/promcfg.go#L1812-L1989) instead of duplicating the logic. The trick would be to map Thanos versions to Prometheus versions based on the Thanos [go.mod](https://github.com/thanos-io/thanos/blob/5f31fc5ad310962f98229617d573f6337eb3ee4c/go.mod#L240) information (e.g. if Thanos vX is pinned to version Y of Prometheus then the config generator should be initialized with version Y).

Reply:
I think thanos config and promethues config are not very well combined, first of all NewConfigGenerator is constructed according to prometheus. I tried to merge it and found that the code coupling is very high.

Reply:
I think thanos config and promethues config are not easy to merge together. First of all, configgenerator is constructed according to prometheus. I tried to merge it and found that the code coupling is very high.